### PR TITLE
ENH: std for dt64 dtype

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/nanops.py [hunk 1]
```diff
@@ -785,6 +784,9 @@ def nanstd(values, axis=None, skipna=True, ddof=1, mask=None):
     >>> nanops.nanstd(s)
     1.0
     """
+    if values.dtype == "M8[ns]":
```
Comment (jreback) [inline pandas/core/nanops.py:787]: shouldn't this be done in _get_values?

File: pandas/tests/arrays/test_timedeltas.py [hunk 2]
```diff
@@ -290,8 +290,18 @@ def test_sum_2d_skipna_false(self):
         )._values
         tm.assert_timedelta_array_equal(result, expected)
 
-    def test_std(self):
-        tdi = pd.TimedeltaIndex(["0H", "4H", "NaT", "4H", "0H", "2H"])
+    # Adding a Timestamp makes this a test for DatetimeArray.std
```
Comment (jreback) [inline pandas/tests/arrays/test_timedeltas.py:293]: shouldn't this be in test_datetimes? yes i read your comment, but still the location is odd

File: pandas/tests/arrays/test_timedeltas.py [hunk 3]
```diff
@@ -303,18 +313,20 @@ def test_std(self):
...
         result = arr.std(skipna=False)
         assert result is pd.NaT
 
         result = tdi.std(skipna=False)
         assert result is pd.NaT
 
-        result = nanops.nanstd(np.asarray(arr), skipna=False)
-        assert result is pd.NaT
+        if getattr(arr, "tz", None) is None:
+            result = nanops.nanstd(np.asarray(arr), skipna=False)
```
Comment (jreback) [inline pandas/tests/arrays/test_timedeltas.py:328]: are we sure the actual result is correct? e.g. you are just comparing vs the implementation itself. IOW a result which checks a fixed series of datetimes for mean/std would be good.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/nanops.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/nanops.py:787]: this is specific to std

File: pandas/tests/arrays/test_timedeltas.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/tests/arrays/test_timedeltas.py:328]: agreed these tests needs fleshing out.  im still giving thought to where they should live

Comment (jreback) [reply pandas/tests/arrays/test_timedeltas.py:328]: sure, that's fine about location. i am more concerned with if the results are actually correct; reasoning is that they are converted to i8 but need to make sure that these are coming back correct as well. i am not sure this is actually tested.

Comment (jbrockmendel) [reply pandas/tests/arrays/test_timedeltas.py:328]: Yes, we have tests with explicit `expected` for std() with td64 in test_reductions, test_describe, and test_analytics.

**Review by jreback — APPROVED** [review]

**Pull request merged**